javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,69 @@
+//Movies exercise
+
+const apiUri = `https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json`;
+
+async function movies() {
+  const response = await fetch(apiUri);
+  const dataMovies = await response.json();
+  console.log(dataMovies);
+
+  const badMovies = dataMovies.filter(function (params) {
+    return params.rating < 4;
+  });
+
+  console.log(badMovies);
+
+  const MoviesSinec = badMovies.filter(function (params) {
+    return params.year === 2000;
+  });
+
+  console.log(MoviesSinec);
+}
+movies();
+
+//Promise that resolves after set time
+
+const afterSeacond = new Promise((resolveAfter) => {
+  setTimeout(() => {
+    resolveAfter("I am the Promise");
+  }, 8000);
+});
+
+afterSeacond.then((elment) => {
+  console.log(elment);
+});
+
+//Rewrite time
```

Comment:
The exercise is to create two functions:
setTimeoutPromise and getCurrentPositionPromise.

The goal is to learn how to wrap callback based APIs into Promise based APIs.

Please create two functions that you can call like this:
```js
setTimeoutPromise(8000).then(() => {
  console.log('I will print after 8 seconds');
});
```

```js
getCurrentPositionPromise().then((position) => {
  console.log(position);
}).catch((error) => {
  console.error(error);
});
```

Inside these functions you'll need to call either `setTimeout` or `getCurrentPosition` and handle the responses of the callbacks.
The functions should return a new Promise, which looks like this:
```js
return new Promise((resolve, reject) => {
  // Call callback API, if it goes well call resolve, if it goes wrong call reject
});
```